feat: Improve TNGFUE execution flow

oliveiraleo opened the pull request:
Closes https://github.com/free5gc/free5gc/issues/614

oliveiraleo committed:
fix: Silence all warnings on wpa_supplicant compilation

The flag helps to reduce the verbosity of the output during compilation time but keeps the error messages untouched in case the user needs them

oliveiraleo committed:
feat: Add a script to automate prerequisite installation

According to the instructions from https://free5gc.org/guide/TNGF/tngfue-installation/ this script installs the required packages and configuration files and builds the wpa_supplicant binary automatically

oliveiraleo committed:
feat: Add a script to deploy the TNGFUE

This script loads the TNGFUE and removes the network interfaces automatically on exit

oliveiraleo committed:
docs: Revamp README content to improve readability

oliveiraleo committed:
docs: Add the TNGFUE new execution flow to README

oliveiraleo committed:
feat(run): Add commands to automate the IP and route configuration

oliveiraleo committed:
fix(run): Fix execution flow with correct paths

free5gc-org commented:
This pull request has been mentioned on **free5GC**. There might be relevant details there:

http://forum.free5gc.org/t/running-tngfue-on-another-subnet/2571/6

oliveiraleo committed:
fix(run): Fix a typo on command in the terminate function

oliveiraleo committed:
fix(run): Increase timer to let wpa_supplicant finish correctly

oliveiraleo committed:
feat(run): Add a debug param to control the script execution flow

The ENABLE_DEBUG parameter is disabled by default and controls if the debug messages are displayed or not during execution

oliveiraleo committed:
fix(run): Update terminate function command order and fix a typo

oliveiraleo committed:
docs: Add the debug parameter to README tips

oliveiraleo committed:
fix(prepare): Give execution permission to the script

oliveiraleo committed:
feat(run): Read IFACE_NAME from sec.conf if not already set

blackcat-118 commented on run.sh:
```diff
@@ -54,6 +54,12 @@ function terminate()
+    IFACE_NAME="$(head -n 1 ./wpa_supplicant/sec.conf)" | cut -d : -f 2 # get the second slice (colon separated) of the first line of the file
+else
+    echo "[INFO][TNGFUE] Interface already set. Using $IFACE_NAME as interface name"
+
```
fi

blackcat-118 commented on run.sh:
```diff
@@ -54,6 +54,12 @@ function terminate()
 
+# If IFACE_NAME is empty, read the Wi-Fi interface name from sec.conf
+if [[ -z $IFACE_NAME ]]; then
+    IFACE_NAME="$(head -n 1 ./wpa_supplicant/sec.conf)" | cut -d : -f 2 # get the second slice (colon separated) of the first line of the file
```
IFACE_NAME="$(head -n 1 ./wpa_supplicant/sec.conf | cut -d : -f 2)" # get the second slice (colon separated) of the first line of the file

blackcat-118 commented on run.sh:
```diff
@@ -0,0 +1,86 @@
+    IFACE_NAME="$(head -n 1 ./wpa_supplicant/sec.conf)" | cut -d : -f 2 # get the second slice (colon separated) of the first line of the file
+else
+    echo "[INFO][TNGFUE] Interface already set. Using $IFACE_NAME as interface name"
+
```
```suggestion
fi

```

blackcat-118 commented on run.sh:
```diff
@@ -0,0 +1,86 @@
+
+# If IFACE_NAME is empty, read the Wi-Fi interface name from sec.conf
+if [[ -z $IFACE_NAME ]]; then
+    IFACE_NAME="$(head -n 1 ./wpa_supplicant/sec.conf)" | cut -d : -f 2 # get the second slice (colon separated) of the first line of the file
```
```suggestion
    IFACE_NAME="$(head -n 1 ./wpa_supplicant/sec.conf | cut -d : -f 2)" # get the second slice (colon separated) of the first line of the file
```

blackcat-118 replied:
`IFACE_NAME=$(head -n 1 ./wpa_supplicant/sec.conf | cut -d : -f 2)`
also works.

oliveiraleo committed:
fix(run): Update if conditional to make code correctly work

Co-authored-by: blackcat-118 <160688822+blackcat-118@users.noreply.github.com>

oliveiraleo committed:
fix(run): Update command formatting according to suggestion

Co-authored-by: blackcat-118 <160688822+blackcat-118@users.noreply.github.com>

blackcat-118 commented:
LGTM!!

andy89923 commented on run.sh:
```diff
@@ -0,0 +1,87 @@
+IFACE_MASK='24' # network mask to be used by configuration command
+IFACE_BROADCAST_IP='192.168.1.255' # IP address to be used by configuration command
+ENABLE_DEBUG=0 # controls if debug messages are enabled
+
```
```suggestion
TNGFUE_PATH=$(pwd)

```

andy89923 commented on run.sh:
```diff
@@ -0,0 +1,87 @@
+    echo "[INFO][TNGFUE] Removing IP and route"
+    sudo ip addr flush $IFACE_NAME # removes IP and default route too
+
+    cd ..
```
```suggestion
    cd $TNGFUE_PATH
```

andy89923 replied:
And I think it's not necessary to cd back.

andy89923 replied:
If so, the `$TNGFUE_PATH` could be remove also.

oliveiraleo replied:
Hello, indeed it's not necessary to cd back.

When I added this cd command I was thinking of designing run.sh in a different way. However after implementing it with the current design this command was left behind and is unnecessary because as soon as the script ends, the shell is back on the path where the script was invoked.

Your suggestions were applied on https://github.com/free5gc/tngfue/pull/2/commits/443e4d123e4ba24fd72110e0743e445a541fb447

Thanks for the review

oliveiraleo committed:
fix(run): Remove unneeded change directory command

This commit applies the suggestion made on https://github.com/free5gc/tngfue/pull/2/files#r1821905555

andy89923 reviewed (approved):
LGTM! 
Thanks for the PR.
